src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -493,11 +498,14 @@ MRAcquisitionData::sort_by_time()
 		vt.push_back( t );
 	}
 
-	if (index_)
-		delete[] index_;
-	index_ = new int[num_acquis];
-	Multisort::sort( vt ,index_ );
+	index_.resize(num_acquis);
+	
+	if( num_acquis <= 0 )
+		std::cerr << "WARNING: You try to sort by time an empty container of acquisition data." << std::endl;
+	else
+		Multisort::sort( vt, &index_[0] );
 
+	sorted_ = true;
```

Comment:
Not a good idea IMHO - methods that compute coil sensitivity maps check whether the acquisitions are sorted **with respect to slices** (at present, on Python/Matlab level, but will be moved down to c++ eventually), rather call it `sorted_by_time_` and add method `sorted_by_time()` (or else add argument to `sorted()` specifying the kind of sort - actually, `sort()` could have it too, instead of your separate sorting method).

Reply:
Agreed, in the `index(int i)` there is not even a check any more if `sorted_== true`, just if `index.size()>0`. So a bool variable to check if it is time sorted is not necessarry atm.